### PR TITLE
Build containerd from guardian

### LINKED ISSUE #252: containerd  and runc are included in two places that cause versions falling out of sync
## Description
Currently we have two versions of containerd under this repo:
1. [Submoduled](https://github.com/cloudfoundry/garden-runc-release/blob/develop/.gitmodules#L45-L48) under `gopath`
2. [Go Module](https://github.com/cloudfoundry/guardian/blob/main/go.mod#L18) under `guardian`

This is causing an issue whenever [there is a bump for the module](https://github.com/cloudfoundry/guardian/pull/343). We would have to make sure that the submodule is also bumped so that the two versions are in sync.

This same situation is also true for `runc`. As for `runc`, it looks like the versions have already diverged, where [submoduled version](https://github.com/cloudfoundry/garden-runc-release/tree/8d86f33aaa7c9cb89833fdc2a2769eb98ff144c1/src/gopath/src/github.com/opencontainers) is lacking [behind moduled version](https://github.com/cloudfoundry/guardian/blob/5e4769bc121bda08bf422cb945be8a38f9474e06/go.mod#L104) in `guardian`.

## Resolution
Modify `guardian` so that required binaries are all checked in under `vendor` directory. This would require changing [packaging spec](https://github.com/cloudfoundry/garden-runc-release/blob/develop/packages/containerd/spec) and [ci helpers](https://github.com/cloudfoundry/garden-runc-release/blob/develop/ci/helpers/build-binaries.bash#L80) so that correct path of the binary is being built. With this change, we would get notified of any changes in upstream and able to patch both the `guardian` module and bosh-release to built the new version.

- [x] containerd
   - cloudfoundry/garden-runc-release#253
   - cloudfoundry/guardian#344
- [x] runc
  - cloudfoundry/garden-runc-release#255
  - cloudfoundry/guardian#356

### PR DESCRIPTION
**Pull request by winkingturtle-vmw**
Dependent PR: cloudfoundry/guardian#344
Resolves cloudfoundry/garden-runc-release#252

### CODE REVIEW

**Review by geofffranks — APPROVED** [review]

File: packages/containerd/packaging [hunk 1]
```diff
@@ -2,15 +2,19 @@
 set -euo pipefail
 
 mkdir -p "${BOSH_INSTALL_TARGET}/src"
-mv gopath/src/* "${BOSH_INSTALL_TARGET}/src"
-mv "${BOSH_INSTALL_TARGET}/src" .
+mv * ${BOSH_INSTALL_TARGET}/src
```
Comment (MarcPaquette) [inline packages/containerd/packaging:5]: That wildcard makes me a bit anxious. Is there more we could add to it?

Comment (winkingturtle-vmw) [reply packages/containerd/packaging:5]: This is the pattern we use on pretty much all of the release we manage. e.g. [bbs](https://github.com/cloudfoundry/diego-release/blob/50ea9cf6abc1b2f82d4732b9953646c0c1ee5aeb/packages/bbs/packaging) and [gorouter](https://github.com/cloudfoundry/routing-release/blob/develop/packages/gorouter/packaging). It gets run in the context of bosh where it's controlled by what files are in the spec.

Comment (MarcPaquette) [reply packages/containerd/packaging:5]: Alright, still seems... icky.

**Review by MarcPaquette — APPROVED** [review]
Changes look good

**Pull request merged**